### PR TITLE
New functions for flk

### PR DESCRIPTION
**Pull request by rosineygp**
Hi,

I just build a lot of addictional functions for flk...
I know it's is not the best way to report but It's has a lot of common and useful function:

"low level"
filter | pmap | println_stderr | exit! | trap! | unset | str-join | str-subs | range | str-ident | mod | file-exists | file-write | unlink

"high level"
or | and | hashmap-list | merge | key-name | call | callable? | get-in | even? | odd? | load-once | spit

I create this functions for my project rosineygp/lines.

I think I will not need any new function for a while ;)

If you need history take a look:
https://github.com/rosineygp/lines/tree/dev/patches
https://github.com/rosineygp/lines/blob/dev/src/includes/lang-utils.clj

### CODE REVIEW

File: docs/flk [hunk 1]
```diff
@@ -54,8 +54,8 @@ _obj_type () {
         func) r="function" ;;
         strn)
             local s="${ANON["${1}"]}"
-            if [[ "${1:0:1}" = "${__keyw}" ]] \
-                || [[ "${1:0:2}" = "${__keyw}" ]]; then
+            if [[ "${s:0:1}" = "${__keyw}" ]] \
```
Comment (rosineygp) [inline docs/flk:57]: It's a bug fix, type never return a keywork, something like this... I don't remember well

File: docs/flk [hunk 2]
```diff
@@ -476,6 +563,12 @@ TOKENIZE () {
     local idx=0
     local chunk=0
     local chunksz=500
+
+    if ((datalen>500))
```
Comment (rosineygp) [inline docs/flk:567]: It's a bug, when the word is too big the parse break... I got it when the read-string need to get something bigger than 500 chars

**Comment by chr15m** [comment]
Looks great, thanks for your contribution! It would be good if there were tests for some of these new functions.

**Comment by rosineygp** [comment]
I didn't write any tests... sorry. (I just create the code and run big integrations "the bad way")

I will not continue to using the language. The final performance of my project is too slow to use in real life, even after I tried some hacks to speed performance. :sob: 

But the main reason for me to try this language was learning about interop and clojure lisp.

I will migrate this code to:
https://joker-lang.org/
or
https://janet-lang.org/

Thanks for everything.

**Comment by chr15m** [comment]
Cool, thanks for this. If somebody adds tests and doc I will merge. Thanks for the useful information about performance and best of luck with your next steps!

**Comment by rosineygp** [comment]
tests are coming!!!

I built a small framework at `tests/test.clj`

execute:

```
./flk tests/lang-utils.clj
./flk tests/patches.clj
```

when it finishes I will notify!

**Comment by chr15m** [comment]
Wow this looks cool. :+1:

**Comment by rosineygp** [comment]
Put test for all included functions (done)

**Comment by chr15m** [comment]
@rosineygp I've added a section to the README called [contributing](https://github.com/chr15m/flk/#contributing) because I realized there are some other things this needs before it can be merged:



 * Bash functions should be added to extras.sh as this is merged from the mal project into the `flk` script at build time. The script should not be directly edited.
 * New flk functions go in src in a new .clj file, and then add the file name to the LOCALMALS list in the Makefile so it gets included in the build.
 * Changes, bugfixes, etc. to mal itself should be submitted upstream (I think your patch includes a few).
 * Document any new functions in README.
 * Unit tests for new functions should go in a .mal file in the tests folder, and be added to the Makefile test clause.

**Comment by rosineygp** [comment]
So... I will break it to small PR.

**Comment by chr15m** [comment]
Sorry to create more work for you. If you want me to do the upstream PR to changes to mal itself I can do that, just let me know which parts come from there.

**Comment by rosineygp** [comment]
> Sorry to create more work for you. If you want me to do the upstream PR to changes to mal itself I can do that, just let me know which parts come from there.

no problem @chr15m, each line of code needs to be placed in right place to work. I am not upset for this.
I will try to upload main fixes to mal.

**Pull request closed without merging**